File: Algorithms/selection sort/selection_sort.c
#include <stdio.h>










void selection_sort(int array[], int size)
{
    // hold the index of the minimum value
    int min_value_index;


    for (int i = 0; i < size-1; i++)
    {
        min_value_index = i;

        // find the minimum value
        for (int j = i+1; j < size; j++)
            if (array[j] < array[min_value_index])
                min_value_index = j;

        
        // swap
        int hold = array[i];
        array[i] = array[min_value_index];
        array[min_value_index] = hold;
    }


    return;
}





// a simple function for printing arrays 
void printArray(int array[], int size)
{
    printf("[");

    for (int i = 0; i < size; i++)
        printf("%d, ", array[i]);

    printf("\b\b]\n");
}






int main()
{
    printf("\n\n ---------------- \n\n");
    printf("a simple test:\n\n");
    

    int array[] = {2, 12, 4, 0, -21, 23, -11, 43, 432, 130, 99, 33};

    printf("before sort: ");
    printArray(array, 12);



    selection_sort(array, 12);

    printf("after sort: ");
    printArray(array, 12);


    printf("\n ---------------- \n\n\n");




    return 0;
}

File: Algorithms/selection sort/SelectionSort.java
import java.util.Arrays;





/**
 * Use {@link SelectionSort#selectionSort(int...)} to sort your {@code int} Array
 * 
 * 
 * @author Mohammad Mahdi Malmasi
 * @version 0.1.0
 */
public class SelectionSort
{


     /**
     * This method sorts your int array with selection sort algorithms
     * 
     * @param array : array to sort
     */
    public static void selectionSort(int...array)
    {
        // hold the index of the minimum value
        int minValueIndex = 0;


        for (int i = 0; i < array.length-1; i++)
        {
            minValueIndex = i;

            // find the minimum value
            for (int j = i+1; j < array.length; j++)
                if (array[j] < array[minValueIndex])
                    minValueIndex = j;


            // swap
            int hold = array[i];
            array[i] = array[minValueIndex];
            array[minValueIndex] = hold;
        }
    }




    /**
     * A simple example for {@link SelectionSort#selectionSort(int...)}
     *  
     * @param args
     */
    public static void main(String[] args) 
    {
        System.out.println("\n\n ---------------- \n");
        System.out.println("a simple test:\n");
        

        int[] array = {8, 10, 25, 0, -34, 23, -11, 43, 50, 123, 7, 33, 12};
        System.out.println("before sort: " + Arrays.toString(array));


        selectionSort(array);
        System.out.println("after sort: " + Arrays.toString(array));


        System.out.println("\n ---------------- \n\n");
    }
}